File: LAST/第七題-Python 繼承/Cuboid.py
from Shape import Shape

class Cuboid(Shape):
    def __init__(self, length, width, height, color):
        self.__length = length
        self.__width = width
        self.__height = height
        super().setColor(color)

    def setLength(self, length):
        self.__length = length
    def setWidth(self, width):
        self.__width = width
    def setHeight(self, height):
        self.__height = height

    def getLength(self):
        return self.__length
    def getWidth(self):
        return self.__width
    def getHeight(self):
        return self.__height

    def __str__(self):
        return "Cuboid"

    def getVolume(self):
        return self.__length * self.__width * self.__height

File: LAST/第四題-C++多型/Cuboid.cpp
#include"Cuboid.h"

Cuboid::Cuboid(double length, double width, double height, string color) 
	:length(length), width(width), height(height){
	setColor(color);
}

void Cuboid::toString() {
	cout << "Cuboid:(" << int(length) << "," << int(width) << "," << int(height) << "),";
}
double Cuboid::getVolume() {
	return width * height * length;
}